### ISSUE TITLE
[no-unused-vars] false positive on styled component props

### ISSUE THREAD

**Issue by geniou** [typescript-eslint/typescript-eslint#2452]
When trying to update from 3.10.1 to 4.0.1 I get false positive for

```TS
type PaymentsProps = {
  isValid: boolean
}

const StyledPayment = styled.div<StyledPaymentProps>``
```

.eslintrc.js

```JSON
    '@typescript-eslint/no-unused-vars': ['error', { argsIgnorePattern: '^_' }],
```

**Expected Result**

I should get no error.

**Actual Result**

I get 

```
  'StyledPaymentProps' is defined but never used  @typescript-eslint/no-unused-vars
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.0.1` |
| `@typescript-eslint/parser`        | `4.0.1` |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | `7.8.0` |
| `node`                             | `12.18.1` |

**Comment by alexkatz** [comment]
Also the case for ```interface``` in addition to ```type```:

```typescript
interface UserRowProps {
  $isEven: boolean;
}

const UserRow = styled.div<UserRowProps>`...`
```

An unused variable is detected here, as well:

```warning  'UserRowProps' is defined but never used  @typescript-eslint/no-unused-vars```

**Comment by Elia-d** [comment]
I'm also having the same issue. It started after upgrading to 4.0.1 😕

**Comment by TrevorSayre** [comment]
This was fixed with https://github.com/typescript-eslint/typescript-eslint/releases/tag/v4.1.0 🎉
